src/Language/PureScript/Errors.hs

```diff
@@ -534,10 +541,14 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
       line $ "The type declaration for " <> markCode (showIdent nm) <> " should be followed by its definition."
     renderSimpleErrorMessage (RedefinedIdent name) =
       line $ "The value " <> markCode (showIdent name) <> " has been defined multiple times"
-    renderSimpleErrorMessage (UnknownName name@(Qualified Nothing (IdentName (Ident i)))) | i `elem` [ C.bind, C.discard ] =
+    renderSimpleErrorMessage (UnknownName name@(Qualified Nothing (IdentName (Ident i))) Nothing) | i `elem` [ C.bind, C.discard ] =
       line $ "Unknown " <> printName name <> ". You're probably using do-notation, which the compiler replaces with calls to the " <> markCode i <> " function. Please import " <> markCode i <> " from module " <> markCode "Prelude"
-    renderSimpleErrorMessage (UnknownName name) =
+    renderSimpleErrorMessage (UnknownName name Nothing) =
       line $ "Unknown " <> printName name
+    renderSimpleErrorMessage (UnknownName name (Just (ty, ctx))) =
+      paras $ [ line $ "Unknown " <> printName name <> " has the inferred type "
+              , markCodeBox (indent (typeAsBox maxBound ty))
+              ] ++ renderContext ctx
```

Comment:
Something about this error message isn't right to me. I know it was just "Unknown foo" before which is also terrible, but adding "has the inferred type" makes it seem less like an error.

Maybe something like for unqualified
> The name foo is not in scope.

> The name foo is not in scope. It has the inferred type ...

And for qualified:
> The name foo is not exported by module Foo.

> The name foo is not exported by module Foo. It has the inferred type ...

@hdgarrood @colinwahl What do you think?

Reply:
I like the idea of having different errors depending on whether or not the name is qualified!  This particular error did seem very terse - I think this is a improvement

Reply:
@natefaubion Actually, now that I think about it, this may be missing a couple of cases.  You could have some declaration
```
baz :: Effect Int
baz = do
  Bar.bar
```
Where the module `Bar` does export a value `bar`, but there just isn't an import for it.  In this case, I don't think that the error `The name bar is not exported by module Foo.` Is an appropriate error.

Maybe something like `The name bar is not imported from module Bar. ...` Would be more accurate.  What do you think?

Reply:
This one has actually bitten me one more than one occasion - not very serious - but it can just nudge you into haring off looking for the export in the imported file instead of fixing the import. If you have any kind of doubts about your build system integrity (as seems to be too often the case) it can pitch you into a worse yak-shaving-wild-goose-chase digression there too.